File: bookhub-front/src/dtos/PageResponseDto.ts
import { AuthorResponseDto } from "./author/response/Author.response.dto";

export interface PageResponseDto<T>{
    map(arg0: (a: AuthorResponseDto) => { label: string; value: number; }): unknown;
    content: T[];
    totalElements: number;
    totalPages: number;
    currentPage: number;
}

File: bookhub-back/src/main/java/com/example/bookhub_back/dto/PageResponseDto.java
package com.example.bookhub_back.dto;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;

import java.util.List;

@Getter
@Builder
@AllArgsConstructor(access = AccessLevel.PRIVATE)
public class PageResponseDto<T> {
    private final List<T> content;
    private final long totalElements;
    private final int totalPages;
    private final int currentPage;

    public static <T> PageResponseDto<T> of(
            List<T> content, long totalElements, int totalPages, int currentPage){
        return PageResponseDto.<T>builder()
                .content(content)
                .totalElements(totalElements)
                .totalPages(totalPages)
                .currentPage(currentPage)
                .build();
    }

}